runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java

```diff
@@ -156,9 +181,11 @@ public static StreamExecutionEnvironment createStreamExecutionEnvironment(
         throw new IllegalArgumentException("The checkpoint interval must be positive");
       }
       flinkStreamEnv.enableCheckpointing(checkpointInterval, options.getCheckpointingMode());
-      flinkStreamEnv
-          .getCheckpointConfig()
-          .setCheckpointTimeout(options.getCheckpointTimeoutMillis());
+      if (options.getCheckpointTimeoutMillis() != -1) {
```

Comment:
nit: < 0

Reply:
-1 in Flink means disabled. Anything else should be applied, even other negative numbers which will raise an exception.